internal/postprocessor/main.go

```diff
@@ -17,54 +17,70 @@ package main
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
-	"io"
 	"io/fs"
 	"io/ioutil"
 	"log"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strings"
 
 	"cloud.google.com/go/internal/gapicgen/generator"
 	"cloud.google.com/go/internal/gapicgen/git"
 	"cloud.google.com/go/internal/gensnippets"
 	"cloud.google.com/go/internal/postprocessor/execv"
 	"cloud.google.com/go/internal/postprocessor/execv/gocmd"
-	"golang.org/x/sync/errgroup"
+	"github.com/google/go-github/v35/github"
+
 	"gopkg.in/yaml.v2"
 )
 
+var hashFromLinePattern = regexp.MustCompile(`.*/(?P<hash>[a-zA-Z0-9]*).*`)
+var firstPartTitlePattern = regexp.MustCompile(`(?P<titleFirstPart>)(\: *\[)(.*)`)
+
+// a more general regex expression for pattern2 would be `.*\] *(?P<titleSecondPart>.*)`
+// but for readability and prevent removal of potentially relevant info the below may be preferable
+var secondPartTitlePattern = regexp.MustCompile(`.*\: *\[REPLACEME\] *(?P<titleSecondPart>.*)`)
+
+var ErrOutOfScope = errors.New("changed files are out of scope")
+var ErrNoScope = errors.New("no changed files found at commit hash")
+var ErrMultiScope = errors.New("multiple scopes found")
+
 func main() {
-	var stagingDir string
-	var clientRoot string
-	var googleapisDir string
-	var directories string
-	flag.StringVar(&stagingDir, "stage-dir", "owl-bot-staging/src/", "Path to owl-bot-staging directory")
-	flag.StringVar(&clientRoot, "client-root", "/repo", "Path to clients")
-	flag.StringVar(&googleapisDir, "googleapis-dir", "", "Path to googleapis/googleapis repo")
-	// The module names are relative to the client root - do not add paths. See README for example.
-	flag.StringVar(&directories, "dirs", "", "Comma-separated list of modules to run")
+	stagingDir := flag.String("stage-dir", "owl-bot-staging/src/", "Path to owl-bot-staging directory.")
```

Comment:
I changed the initialization of these to be consistent with other areas in our repo